protocol/strict-derivation.md

```diff
@@ -0,0 +1,269 @@
+
+At this point, there are no open questions about the design of this protocol change. I assume that
+the actual implementation will raise questions on certain details.
+
+# Activation rules & actions
+
+I propose to activate the new rules by L1 inclusion block timestamp, which is most convenient and clean
+for derivation protocol upgrades. It's accessible in most parts of the derivation pipeline as the
+stage's _origin_ (not to be confused with the _L1 origin of an L2 block_).
+Note that this is in contrast to how Delta activated span batches, which was by first parsing the
+span batch and then checking the span batch's L1 origin timestamp.
+
+However, we need to specify how existing buffered frames, channels and batches are treated at
+activation, since the stricter rules of Holocene take effect.
+To start with a derivation pipeline state that is consistent with Holocene rules, I propose the
+following special activation effects to take place when the pipeline's origin reaches the activation
+timestamp, which is similar to applying Holocene rules retroactively.
+- Channel Bank:
+  1. All channels that don't consist of contiguous frames that include the first, are dropped. 
+  2. All but the youngest (in terms of L1 inclusion) channel are dropped.
+  3. This optional left-over channel
+     will be the start staging channel for the Holocene channel bank.
+
+The principle of eager derivation of later stages should guarantee at this point that all batches that could have
+been applied directly on top of the current safe chain have already been exhausted from the batch
+queue. So what is left should be `undecided` and `future` batches. (Note: I'm not sure on this one.)
+I see two options on how to handle a non-empty batch queue at this point:
+- Option 1: Drop future batches, continue resolving undecided batches, if any are left, and apply
+new Holocene rules.
+- Option 2: The Batch Queue will just start applying Holocene rules from this moment onwards. This will then
+automatically derive gaps as empty batches, discard out of order batches, etc. A problem with this
+option is that the sync start for a while after Holocene would still need to apply the pre-Holocene
+algorithm to restore the derivation pipeline state.
+
+I'm undecided at this point which option is better. I'd prefer to pick the one that leads to a
+simpler implementation, minimizing complexity where possible. Since almost always a healthy batcher
+implementation during normal network conditions will already produce batches according to Holocene
+rules, we don't need to optimize for the prettiest activation rules, as long as they work
+consistently.
+
+Another open question is how to handle span batches that come from pre-Holocene channels.
+I propose that, for simplicity of implementation, if they are found to be valid as a span batch, to
+just apply the new Partial Span Batch Validity rules even though those span batches were derived
+from pre-Holocene L1 blocks.
```

Comment:
This would only apply to channels that span the Holocene activation right? Anything full pre-holocene would use the old rules and anything full post-Holocene would use the new rules.  If so I agree, though it wouldn't be an issue if we discarded any incomplete channels at Holocene activation.

Reply:
It would technically also apply to full pre-Holocene span batches, if the channel that contained this span batch was already closed before Holocene activation, but included on L1 only after Holocene activation.

Since correct batcher implementations don't violate this property anyways, I just want to pick the rule with the lowest implementation lift.